### PR TITLE
AK+LibCore: `StringView(char const[N])` and `StringView` in `ArgsParser`

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
### AK: Allow StringView to be constructed from a `char const[N]`

This also includes a lot of changes on AK::URL users, as those were
relying on the implicit conversion from `char const*` to `ByteString`
to `StringView` to work, is now ambiguous
Similar issues for `RIFF::ChunkID`

### LibCore: Use `StringView`s in ArgsParser

StringView is a lot safer than raw pointers, which we tend to avoid.
This code was now around 2yo so time to modernize it a bit.

This has a lot of `nullptr` -> `{}` changes, which seems as ambiguous as
before

### CODE REVIEW

File: Userland/Shell/Builtin.cpp [hunk 1]
```diff
@@ -1651,7 +1651,7 @@ ErrorOr<int> Shell::builtin_argsparser_parse(Main::Arguments arguments)
         .value_name = "string",
         .accept_value = [&](StringView value) {
             VERIFY(strlen(value.characters_without_null_termination()) == value.length());
```
Comment (Hendiadyoin1) [inline Userland/Shell/Builtin.cpp:1653]: @alimpfard Is this check still relevant now?

Comment (alimpfard) [reply Userland/Shell/Builtin.cpp:1653]: No.

File: AK/StringView.h [hunk 2]
```diff
@@ -22,6 +22,13 @@ namespace AK {
 class StringView {
 public:
     ALWAYS_INLINE constexpr StringView() = default;
+    template<size_t N>
```
Comment (trflynn89) [inline AK/StringView.h:25]: Please put newlines around this ctor. I know there aren't newlines around the others, but let's not make an ugly situation uglier.

I think these should also be 2 different commits.
Commit 1 - use StringView in the URL ctors
Commit2 - add the consteval StringView ctor

Comment (trflynn89) [reply AK/StringView.h:25]: Alternatively, I don't know why we have a `URL(ByteString)` ctor. Seems that can just be removed too - if someone passes a `ByteString` it can be implicitly converted to a `StringView` and hit the `URL(StringView)` ctor.

File: Userland/Libraries/LibCore/ArgsParser.cpp [hunk 3]
```diff
@@ -65,9 +65,9 @@ bool ArgsParser::parse(Span<StringView> arguments, FailureBehavior failure_behav
 
     for (size_t i = 0; i < m_options.size(); i++) {
         auto& opt = m_options[i];
-        if (opt.long_name) {
+        if (!opt.long_name.is_null()) {
```
Comment (trflynn89) [inline Userland/Libraries/LibCore/ArgsParser.cpp:68]: I think we're trying to actively avoid new users of the null state. `Optional<StringView>` instead?

File: AK/StringView.h (hunk 2 above)
Comment (trflynn89) [reply AK/StringView.h:25]: Also let's add a unit test for this ctor

File: Userland/Libraries/LibCore/ArgsParser.cpp (hunk 3 above)
Comment (alimpfard) [reply Userland/Libraries/LibCore/ArgsParser.cpp:68]: +1, please don't use the null state in new code.
